website/content/docs/concepts/plugins/storage/host-volumes.mdx

```diff
@@ -0,0 +1,387 @@
+use the [delete API][api-delete]).
+
+Nomad also runs `delete` when an initial volume `create` operation
+fails, since it may have been only partially completed.
+
+**CLI Arguments:** `$1=delete`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=delete
+DHV_HOST_PATH=/suggested/path/to/place/volumes
+DHV_NODE_ID={Nomad node ID}
+DHV_NODE_POOL={Nomad node pool}
+DHV_NAMESPACE={volume namespace}
+DHV_VOLUME_NAME={name from the volume specification}
+DHV_VOLUME_ID={volume ID assigned by Nomad}
+DHV_PARAMETERS={json of parameters from the volume spec, or: null}
+DHV_PLUGIN_DIR={path to directory containing plugins}
+```
+
+**Expected stdout:** none (stdout is discarded)
+
+**Requirements:**
+
+* Must complete within 60 seconds, or Nomad will kill it
+* Must be idempotent - repeat runs produce the same result
+* Will be run after a partial (failed) create operation TODO TODO TODO !!! how about create=UPDATE ???
+* Must be safe to run concurrently, per volume name per node
+
+</blockquote>
+
+## Considerations
+
+Plugin authors should consider these details when writing plugins.
+
+### Execution
+
+* The plugin is executed as the same user as the `nomad` agent (likely root).
+* Plugin `stdout` and `stderr` are exposed as client agent debug logs,
+  so plugins should not output sensitive information.
+* Nomad does not retry automatically on error. The caller of create/delete
+  must retry manually. The plugin may do so internally with its own retry
+  logic, provided it still completes within the deadline.
+* Errors from `create` while restoring a volume during Nomad agent start
+  do not halt the client. The error will be in client logs, and the volume
+  is not registered as available on the node.
+
+### Uniqueness
+
+* Volume `name` is unique per _node_, and volume `ID` is unique per _region_.
+* Only one create/delete operation at a time is executed per volume `name`
+  per node, and similarly by `id` on Nomad servers, but many create/delete
+  operations for different volume IDs may run concurrently, even on the same
+  node.
+* Plugins that write into network storage need to take care not to delete
+  remote/shared state by `name`, unless they know that there are no other
+  volumes with workloads using that name.
+
+### Configuration
+
+* Per-_volume_ configuration should be set in the volume specification file's
+  `parameters`. Per-_node_ configuration should be put in config file(s) as
+  described next.
+* There is no mechanism built into Nomad for plugin configuration. As a
+  convention, we suggest placing any necessary configuration file(s) next to
+  the executable plugin in the plugin directory. You may use `DHV_PLUGIN_DIR`
+  to refer to the directory.
+* If a plugin needs to retain state across operations (e.g. delete needs
+  some value that was generated during create), then you may store that on
+  the host filesystem, or some external data store of your choosing, perhaps
+  even Nomad variables.
+
+[dhv]: TODO(1.10.0)?
```

Comment:
The top-level DHV documentation is currently just the volume spec page. That holds most of the conceptual info.

Reply:
should this later point to `.../stateful-workloads#dynamic-host-volumes` (or similar) whenever that's written up?

the volume spec is very important, but doesn't quite clearly describe the holistic feature near the top, as it's more of a reference, so I hesitate to link it as the main source of high-level info. of course I do link to it when referring directly to the volume spec.

Reply:
> should this later point to .../stateful-workloads#dynamic-host-volumes (or similar) whenever that's written up?

Probably, yeah. For CSI it's _this_ document in Concepts > Plugins that is the top-level conceptual document. So there's a bit of divergence here because in CSI we're not responsible for defining the plugin spec at all.